core/src/OpenMPTarget/Kokkos_OpenMPTarget_Parallel.hpp

```diff
@@ -81,8 +466,12 @@ class ParallelFor<FunctorType, Kokkos::RangePolicy<Traits...>,
     const typename Policy::member_type begin = m_policy.begin();
     const typename Policy::member_type end   = m_policy.end();
 
-#pragma omp target teams distribute parallel for map(to : this->m_functor)
-    for (int i = begin; i < end; i++) m_functor(i);
+    if (end <= begin) return;
+
+    FunctorType a_functor(m_functor);
```

Comment:
Why do you make a copy?  Is there an issue related to using directly the data member?

Reply:
We need to figure out how to do this with all the compilers so that it actually works.